src/vos/vos_query.c

```diff
@@ -322,28 +337,51 @@ vos_obj_query_key(daos_handle_t coh, daos_unit_oid_t oid, uint32_t flags,
 		return -DER_INVAL;
 	}
 
+	query.qt_ts_set = NULL;
+
 	if (flags & VOS_GET_DKEY) {
 		if (dkey == NULL) {
 			D_ERROR("dkey can't be NULL with VOS_GET_DKEY\n");
 			return -DER_INVAL;
 		}
 		daos_anchor_set_zero(&dkey_anchor);
+
+		cflags = VOS_TS_READ_OBJ;
 	}
 
 	if (flags & VOS_GET_AKEY && akey == NULL) {
 		D_ERROR("akey can't be NULL with VOS_GET_AKEY\n");
 		return -DER_INVAL;
+
+		if (cflags == 0)
+			cflags = VOS_TS_READ_DKEY;
```

Comment:
Hmm... When will this if statement be executed, given the return statement right above?

Reply:
good question.   Not sure what I'm doing here......

Reply:
Ok, I see what happened.   It's essentially a copy paste but didn't read the if statement properly.   Will fix it.   Thanks.